common/http.go

```diff
@@ -0,0 +1,79 @@
+package common
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"gopkg.in/go-playground/validator.v9"
+
+	"github.com/pkg/errors"
+)
+
+type ErrorResponse struct {
+	Success bool `json:"success,omitempty"`
+
+	// Error is the error message if Success is false
+	Error string `json:"error,omitempty"`
+
+	// Code is set if Success is false
+	Code int `json:"code,omitempty"`
+}
+
+var errNilBody = errors.Errorf("expecting a non-nil body")
+
+// ParseRequestJSON unmarshals into save, the body of the
+// request. It closes the body of the request after parsing.
+// Since it uses json.Unmarshal, save must be of a pointer type
+// or compatible with json.Unmarshal.
+func ParseRequestJSON(r *http.Request, save interface{}) error {
+	if r == nil || r.Body == nil {
+		return errNilBody
+	}
+	defer r.Body.Close()
+
+	slurp, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return errors.Wrap(err, "Read Request")
+	}
+	if err := json.Unmarshal(slurp, save); err != nil {
+		return errors.Wrap(err, "Parse")
+	}
+	return nil
+}
+
+func ParseAndValidateRequestJSON(r *http.Request, save interface{}) error {
+	if err := ParseRequestJSON(r, save); err != nil {
+		return err
+	}
+	return validate(save)
+}
+
+var theValidator = validator.New()
+
+func validate(obj interface{}) error {
+	return errors.Wrap(theValidator.Struct(obj), "Validate")
+}
+
+func WriteSuccess(w http.ResponseWriter, data interface{}) {
+	WriteCode(w, data, 200)
+}
+
+func WriteCode(w http.ResponseWriter, out interface{}, code int) {
+	blob, err := json.MarshalIndent(out, "", "  ")
+	if err != nil {
+		WriteError(w, err)
+	} else {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(code)
+		w.Write(blob)
+	}
+}
+
+func WriteError(w http.ResponseWriter, err error) {
+	resp := &ErrorResponse{
+		Code:  http.StatusBadRequest,
```

Comment:
We do want a way to set this, right? 

How about allowing some special interface with a method `HTTPCode() int` and if err implements that, use it, otherwise BadRequest